File: Eshop.Domain/ValueObjects/Address.cs
namespace Eshop.Domain.ValueObjects
{
    public class Address : ValueObject
    {
        public string Street { get; private set; }
        public string City { get; private set; }
        public string State { get; private set; }
        public string ZipCode { get; private set; }
        public string Country { get; private set; }

        public Address(string street, string city, string state, string zipCode, string country)
        {
            Street = street;
            City = city;
            State = state;
            ZipCode = zipCode;
            Country = country;
        }

        protected override IEnumerable<object> GetAtomicValues()
        {
            yield return Street;
            yield return City;
            yield return State;
            yield return ZipCode;
            yield return Country;
        }
    }
}

File: Eshop.Frontend/eshop.manager/src/components/Shared/Address.js
import React from "react";

const Address = ({ address }) => {
  return address !== undefined ? (
    <>
      <h5>Address:</h5>
      <p>{address.street},</p>
      <p>
        {address.zipCode}, {address.city},
      </p>
      <p>{address.country}</p>
    </>
  ) : (
    <></>
  );
};

export default Address;
